### PR TITLE
A few typo fixes and minor changes in the problem formulation document

### PR DESCRIPTION
**Pull request by josipkh**
This PR fixes a few typos noticed after #1221.
Also, there are some minor modifications suggested to improve readability.
Thanks again for aligning the interfaces and the docs!

### CODE REVIEW

File: docs/problem_formulation/problem_formulation_ocp_mex.tex [hunk 1]
```diff
@@ -309,7 +312,7 @@ \subsection{Cost module \code{'AUTO'} \label{sec:cost:auto}}
 Set \code{AcadosOcpCost.cost\_type\_0}, \code{AcadosOcpCost.cost\_type}, \code{AcadosOcpCost.cost\_type\_e} to \code{'AUTO'} in order to use this cost module.
 In this case \acados{} detects if the cost function specified is a linear least squares term and transcribes it in the corresponding form.
 Otherwise, it is formulated using the external cost module.
-Note: slack penalties are optional and we plan to detected them from the expressions in future versions.
+Note: slack penalties are optional and we plan to detect them from the expressions in future versions.
```
Comment (FreyJo) [inline docs/problem_formulation/problem_formulation_ocp_mex.tex:315]: I think this is not planned anymore, and we should remove this sentence altogether.

**Review by FreyJo — APPROVED** [review]
Thanks a lot for the feedback @josipkh !
Happy that you like the latest changes :)
Maybe @sandmaennchen can check too and merge?

File: docs/problem_formulation/problem_formulation_ocp_mex.tex (hunk 1 above)
Comment (josipkh) [reply docs/problem_formulation/problem_formulation_ocp_mex.tex]: understood, done :)

**Review by sandmaennchen — APPROVED** [review]
thanks! :+1:

**Pull request merged**